src/varity/ref_gene.clj

```diff
@@ -208,28 +208,49 @@
 
+(defn exon-ranges->intron-ranges
+  [exon-ranges]
+  (mapv (fn [[left right]]
+          [(inc (second left)) (dec (first right))])
+        (partition 2 1 exon-ranges)))
+
 (defn seek-gene-region
   "Seeks chr:pos through exon entries in refGene and returns those indices"
   ([chr pos rgidx]
    (seek-gene-region chr pos rgidx nil))
   ([chr pos rgidx name]
-   ;; TODO seek intron region
-   ;; TODO seek UTR-5 or UTR-3 region
    (->> (if name
           (ref-genes name rgidx)
           (ref-genes chr pos rgidx))
         (map (fn [rg]
-               (let [exon-ranges ((case (:strand rg)
-                                    :forward identity
-                                    :reverse reverse) (:exon-ranges rg))
-                     idx (->> exon-ranges
-                              (keep-indexed (fn [i [s e]] (if (<= s pos e) i)))
-                              first)]
-                 {:exon-index (if idx
-                                (inc idx) ; 1-origin
-                                nil)
-                  :exon-count (count exon-ranges)
-                  :gene rg}))))))
+               (let [sgn (case (:strand rg)
+                           :forward identity
+                           :reverse reverse)
+                     exon-ranges (:exon-ranges rg)
+                     exon-idx (->> (sgn exon-ranges)
+                                   (keep-indexed (fn [i [s e]] (if (<= s pos e) i)))
+                                   first)
+                     intron-ranges (if-not exon-idx
+                                       (->> exon-ranges
+                                            (exon-ranges->intron-ranges)
+                                            (sgn)))
+                     intron-idx (->> intron-ranges
+                                     (keep-indexed (fn [i [s e]] (if (<= s pos e) i)))
+                                     first)
+                     utr (cond
+                           (< pos (:cds-start rg)) (first (sgn '({:region "UTR-5" :index 0 :count 1}
+                                                                 {:region "UTR-3" :index 0 :count 1})))
+                           (> pos (:cds-end rg)) (second (sgn '({:region "UTR-5" :index 0 :count 1}
+                                                                {:region "UTR-3" :index 0 :count 1})))
+                           :else nil)
+                     exon-intron (cond
+                                   exon-idx {:region "exon" :index exon-idx :count (count exon-ranges)}
+                                   intron-idx {:region "intron" :index intron-idx :count (count intron-ranges)})
+                     region-types (remove nil? (vector utr exon-intron))]
+                 (update {:region-types region-types :gene rg}
+                         :region-types
+                         (partial map #(update % :index (fn [idx]
+                                                          (when idx (inc idx))))))))))))
```

Comment:
I think it's a bit tricky description. I want to keep it simple here, as follows.


```suggestion
                 {:regions regions :gene rg}))))))
```

Then, I think it's easier to understand if you apply `inc` to `index` in the `exon-intron` at line [246 to 248](https://github.com/chrovis/varity/pull/36/files#diff-63a0f54cbd3574ddea6aa13e18ee93d0R246-R248).

```clj
                     exon-intron (cond
                                   exon-idx {:region "exon" :index (inc exon-idx) :count (count exon-ranges)}
                                   intron-idx {:region "intron" :index (inc intron-idx) :count (count intron-ranges)})
```

Besides, sorry to be so detailed, but I prefer `regions` to `region-types`.

Reply:
Thank you for your review :pray:
I have reflected your comment.